### PR TITLE
Make the Vulkan window return a configured GrVkBackendContext so GrContext::Create succeeds.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

**Comment by jason-simmons** [comment]
LGTM

**Comment by chinmaygarde** [comment]
I can't land this just yet. A few patches in Skia have to land first.

File: tools/gn [hunk 1]
```diff
@@ -47,6 +47,7 @@ def to_gn_args(args):
     gn_args = {}
 
     gn_args['is_debug'] = args.unoptimized
+    gn_args['android_full_debug'] = args.target_os == 'android' and args.unoptimized
```
Comment (jason-simmons) [inline tools/gn]: break this out into a separate PR?

File: vulkan/vulkan_window.cc [hunk 2]
```diff
@@ -409,11 +433,31 @@ bool VulkanWindow::CreateCommandPool() {
...
+SkISize VulkanWindow::SurfaceSize() const {
+  VkSurfaceCapabilitiesKHR capabilities = {0};
+  if (vk.getPhysicalDeviceSurfaceCapabilitiesKHR(physical_device_, surface_,
+                                                 &capabilities) != VK_SUCCESS) {
+    return SkISize::Make(0, 0);
+  }
+
+  VkExtent2D extents = capabilities.currentExtent;
+
+  if (extents.width < capabilities.minImageExtent.width) {
```
Comment (jason-simmons) [inline vulkan/vulkan_window.cc]: use std::clamp here?

File: tools/gn (hunk 1 above)
Comment (chinmaygarde) [inline tools/gn]: Will do.

File: vulkan/vulkan_window.cc (hunk 2 above)
Comment (chinmaygarde) [inline vulkan/vulkan_window.cc]: Will do. Didn't even know that was a thing.

**Comment by abarth** [comment]
We should land the skia roll separately.  The skia roll is also going to require updating skia in fuchsia, which I can help you with.

**Comment by chinmaygarde** [comment]
> We should land the skia roll separately. The skia roll is also going to require updating skia in fuchsia, which I can help you with.

My bad. This PR has long been unsalvageable. I thought I closed it. Will land smaller patches separately.
